### PR TITLE
Tour db model 186040769

### PR DESCRIPTION
**Pull request by hannahb-cogapp**

### CODE REVIEW

File: app/Models/CustomTour.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class CustomTour extends Model
+{
+    use HasFactory;
+
+    protected $connection = 'tours_db';
+
+    protected $table = 'tours';
```
Comment (nikhiltri) [inline app/Models/CustomTour.php]: Give the table and class the same name, so you don't have to override the default here.

File: database/migrations/2023_09_25_113832_create_tours_table.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class () extends Migration {
+    protected $connection = 'tours_db';
+
+    public function up(): void
+    {
+        Schema::connection($this->connection)->create('tours', function (Blueprint $table) {
```
Comment (nikhiltri) [inline database/migrations/2023_09_25_113832_create_tours_table.php]: With the `$connection` property set above, you shouldn't need to pass the connection variable to the `Schema::connection()` method here. It should pull that behind the scenes.

File: routes/web.php [hunk 3]
```diff
@@ -218,6 +219,11 @@
 Route::get('/interactive-features/{slug}', [InteractiveFeatureExperiencesController::class, 'show'])->name('interactiveFeatures.show');
 Route::get('/interactive-features/kiosk/{slug}', [InteractiveFeatureExperiencesController::class, 'show'])->name('interactiveFeatures.show-kiosk');
 
+// Tour routes
+Route::get('/tours/builder', [CustomTourController::class, 'build']);
+Route::get('/tours/{id}', [CustomTourController::class, 'show']);
+Route::post('/api/tour', [CustomTourController::class, 'store']);
```
Comment (nikhiltri) [inline routes/web.php]: Can you share the thinking behind not keeping this path in the `/tours` namespace?

File: app/Models/CustomTour.php [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class CustomTour extends Model
+{
+    use HasFactory;
```
Comment (lukew-cogapp) [inline app/Models/CustomTour.php]: It hasn't have a factory

File: app/Models/CustomTour.php [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class CustomTour extends Model
+{
+    protected $connection = 'tours';
```
Comment (tristanr-cogapp) [inline app/Models/CustomTour.php]: Add conditional environment test as in `database/migrations/2023_09_25_113832_create_tours_table.php`

**Review by tristanr-cogapp — CHANGES_REQUESTED** [review]
All good apart from one update to the Model

File: routes/web.php (hunk 3 above)
Comment (tristanr-cogapp) [reply routes/web.php]: Now under /custom-tours/

File: app/Models/CustomTour.php (hunk 5 above)
Comment (hannahb-cogapp) [reply app/Models/CustomTour.php]: Thanks, I've added this

**Review by tristanr-cogapp — APPROVED** [review]

**Pull request merged**